eng/testing/tests.mobile.targets

```diff
@@ -11,6 +11,11 @@
     <UseDefaultBlazorWASMFeatureSwitches Condition="'$(UseDefaultBlazorWASMFeatureSwitches)' == ''">true</UseDefaultBlazorWASMFeatureSwitches>
     <UseDefaultAndroidFeatureSwitches Condition="'$(UseDefaultAndroidFeatureSwitches)' == ''">true</UseDefaultAndroidFeatureSwitches>
     <UseDefaultiOSFeatureSwitches Condition="'$(UseDefaultiOSFeatureSwitches)' == ''">true</UseDefaultiOSFeatureSwitches>
+    <TargetOS Condition="'$(TargetOS.ToLower())' == 'ios'">iOS</TargetOS>
+    <TargetOS Condition="'$(TargetOS.ToLower())' == 'iossimulator'">iOSSimulator</TargetOS>
+    <TargetOS Condition="'$(TargetOS.ToLower())' == 'tvos'">tvOS</TargetOS>
+    <TargetOS Condition="'$(TargetOS.ToLower())' == 'tvossimulator'">tvOSSimulator</TargetOS>
+    <TargetOS Condition="'$(TargetOS.ToLower())' == 'maccatalyst'">MacCatalyst</TargetOS>
```

Comment:
Is this already done for Android and Browser (and others?) somewhere?  Should we do it for all of them?

Reply:
I think this might be iOS only.  And I wonder if it's because of https://github.com/dotnet/runtime/blob/60d169bdec5b876fcfe009b140467a8b706c9bd8/src/tasks/AppleAppBuilder/Xcode.cs#L16-L38 ?

Reply:
IIRC MSBuild shouldn't be case sensitive, so I believe the problem is somewhere else as @steveisok points out.

Reply:
MSBuild is not case sensitive in its conditions. The fix above is definitely not necessary.

Reply:
Thanks, addressed